joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala

```diff
@@ -1,130 +1,667 @@
+
+          capturedClosureBinding.outE.collectAll[Ref].l match {
+            case List(outRef) =>
+              val capturedParam = outRef.inNode.asInstanceOf[MethodParameterIn]
+              capturedParam.name shouldBe "captured"
+              capturedParam.method.head.fullName shouldBe "TestClass.test:Foo(java.lang.String)"
+            case result => fail(s"Expected single REF outE from closureBinding to capturedParam but got $result")
+          }
+
+          capturedClosureBinding.inE.collectAll[Capture].l match {
+            case List(capture) =>
+              val outMethod = capture.outNode().asInstanceOf[MethodRef]
+              outMethod.methodFullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Float)"
+            case result => fail(s"Expected single CAPTURE inE but got $result")
+          }
+
+        case result => fail(s"Expected 2 closure bindings for captured variables but got $result")
+      }
     }
 
-    "should contain a LOCAL node for the captured `diff`" in {
-      // The code for the `getComparator` local is `int diff`, so
-      // this pattern matches only the local created for the lambda.
-      pendingUntilFixed(cpg.local.code("diff").size shouldBe 1)
+    "create a typeDecl node inheriting from correct interface" in {
+      cpg.typeDecl.name(".*lambda.*").l match {
+        case List(lambdaDecl) =>
+          lambdaDecl.name shouldBe "lambda$0"
+          lambdaDecl.fullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Float)"
+          lambdaDecl.inheritsFromTypeFullName should contain theSameElementsAs List("Foo")
+
+        case result => fail(s"Expected a single typeDecl for the lambda but got $result")
+      }
     }
 
-    "should contain a CLOSURE_BINDING node for the captured `diff`" in {
-      pendingUntilFixed(cpg.all.count(_.isInstanceOf[ClosureBinding]) shouldBe 1)
+    // TODO: Fix typeDecl for interfaceBinding
+    "create bindings to implemented method" ignore {
+      cpg.all.collectAll[Binding].nameExact("baz").sortBy(_.methodFullName).toList match {
+        case List(interfaceBinding, erasedBinding, binding) =>
+          interfaceBinding.methodFullName shouldBe "Foo.baz:java.lang.String(java.lang.Object,java.lang.Object)"
+          interfaceBinding.signature shouldBe "java.lang.String(java.lang.Object,java.lang.Object)"
+          binding.inE.collectAll[Binds].map(_.outNode().asInstanceOf[TypeDecl]).l match {
+            case List(typeDecl) => typeDecl.fullName shouldBe "Foo"
+            case result         => fail(s"Expected Binds edge from typeDecl but got $result")
+          }
+
+          binding.methodFullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Float)"
+          binding.signature shouldBe "java.lang.String(java.lang.Integer,java.lang.Float)"
+          binding.inE.collectAll[Binds].map(_.outNode().asInstanceOf[TypeDecl]).l match {
```

Comment:
maybe it's possible to not do the hard cast there

Reply:
You're right! I'm not sure why I chose to do it this way, but it's much nicer to match the specific type instead.

Reply:
Also done